storage/replica_command.go

```diff
@@ -1359,6 +1359,18 @@ func (r *Replica) splitTrigger(batch engine.Engine, split *roachpb.SplitTrigger)
 			// Our in-memory state has diverged from the on-disk state.
 			log.Fatalf("failed to update Store after split: %s", err)
 		}
+
+		// To avoid leaving the new range unavailable as it waits to elect
+		// its leader, one (and only one) of the nodes should start an
+		// election as soon as the split is processed. For simplicity, we
+		// choose the first node in the replica list. If this node is
+		// unavailable, the group will have to wait for an election
+		// timeout, just as with any other leader failure. (we could
+		// improve this by e.g. choosing the node that had the leader
+		// lease before the split and is therefore known to be up)
+		if r.store.StoreID() == split.NewDesc.Replicas[0].StoreID {
+			r.store.multiraft.Campaign(split.NewDesc.RangeID)
```

Comment:
The second commit in this branch attempts to address this issue. It avoids the problem of dropping and recreating the raft group, but it creates another similar problem because the range metadata (replicasByKey) in the Store is no longer updated atomically. I think this could be solved with some more work, but I'm not sure it's worth it, so I'm probably going to back to hacking around the problem just enough to make this test reliable.